core/src/main/scala/org/apache/spark/util/random/RandomSampler.scala

```diff
@@ -155,6 +171,28 @@ class BernoulliSampler[T: ClassTag](fraction: Double) extends RandomSampler[T, T
 
   override def setSeed(seed: Long): Unit = rng.setSeed(seed)
 
+  private val gapSampling: GapSampling = if (fraction > 0.0 && fraction < 1.0) {
```

Comment:
Would this (and subsequent one) maybe be simpler as a lazy val and then get rid of the if/else/null thing? It seems like right now we make this some times even when it isn't used.